private_gpt/components/embedding/embedding_component.py

```diff
@@ -28,9 +28,33 @@ def __init__(self, settings: Settings) -> None:
                         "Local dependencies not found, install with `poetry install --extras embeddings-huggingface`"
                     ) from e
 
+                # Get the number of available GPUs
+                num_gpus = torch.cuda.device_count()
+
+                if num_gpus > 0:
+                    print("Available CUDA devices:")
+                    for i in range(num_gpus):
+                        print(f"GPU {i}: {torch.cuda.get_device_name(i)}")
+                else:
+                    print("No CUDA devices available. Switching to CPU.")
+
+                # Check if CUDA is available
+                if torch.cuda.is_available():
+                    # If settings.embedding.gpu is specified, use that GPU index
+                    if hasattr(settings, 'embedding') and hasattr(settings.embedding, 'gpu'):
+                        gpu_index = settings.embedding.gpu
+                        device = torch.device(f"cuda:{gpu_index}")
+                    else:
+                        # Use the default GPU (index 0)
+                        device = torch.device("cuda:0")
+                else:
+                    # If CUDA is not available, use CPU
+                    device = torch.device("cpu")
```

Comment:
What happens with laptops using a GPU that is not Nvidia based? For example Mac book running Metal GPU? Will this make embedding slower forcing them to go to CPU?

Reply:
This logic looks similar this: [llama_index.core.utils.infer_torch_device](https://github.com/run-llama/llama_index/blob/5c53f41712785e5558156372bdc4f33a6326fa5f/llama-index-core/llama_index/core/utils.py#L463) which handles Metal (mps).